src/webpki.rs

```diff
@@ -175,6 +175,27 @@ impl <'a> EndEntityCert<'a> {
                                  intermediate_certs, &self.inner, time, 0)
     }
 
+    /// Verifies that the end-entity certificate is valid for use by a TLS
+    /// client.
+    ///
+    /// `supported_sig_algs` is the list of signature algorithms that are
+    /// trusted for use in certificate signatures; the end-entity certificate's
+    /// public key is not validated against this list. `trust_anchors` is the
+    /// list of root CAs to trust. `intermediate_certs` is the sequence of
+    /// intermediate certificates that the client sent in the TLS handshake.
+    /// `cert` is the purported end-entity certificate of the client. `time` is
+    /// the time for which the validation is effective (usually the current
+    /// time).
+    pub fn verify_is_valid_tls_client_cert(
+            &self, supported_sig_algs: &[&SignatureAlgorithm],
+            trust_anchors: &[TrustAnchor],
+            intermediate_certs: &[untrusted::Input], time: time::Timespec)
```

Comment:
In general, this seems reasonable. However, I think we need to do a few more things:

First, we need a way to get the client's identity out of the certificate. Some applications might consider "Certificate was issued from a CA I trust" a way of determining "the client is in a trusted group." But in the end people usually need to know more specifically *who* (or what machine, what server, etc.) the client is.

In the TLS server case we pass in the domain name which serves as the identity to match. But with client certificates it isn't so simple. First, there could be multiple names within the client certificate. Secondly, the set of allowed clients for a connection might be large and hard to access (e.g. in an LDAP database), so we can't pass in the list of trusted identities. AFAICT, instead we need a way to loop through the names (distinguished names, or email addresses, or dNSNames) in the certificate, pass each one to a caller-supplied callback, and return success if/when the callback says that there was a match.

Secondly, we need a way to clearly and safely partition sets of trusted root TLS server certificates from sets of trusted root client certificates. For example, we could have something like this:
```rust
struct TLSServerTrustAnchors(&TrustAnchor]);
struct TLSClientTrustAnchors(&[TrustAnchor]);
```
Then crates like webpki-roots would publish their trust anchor sets as a value of type `TLSServerTrustAnchors` instead of `&[TrustAnchor]`.

There are probably some other things to consider, but these are the main issue I see immediately from looking at this.